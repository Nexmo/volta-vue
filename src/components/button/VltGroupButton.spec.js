import { mount } from '@vue/test-utils';
import { VltGroupButton } from '..';

describe('vlt-group-button', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(VltGroupButton, {
      propsData: {
        options: ['red', 'yellow', 'blue'],
      },
    });
  });

  test('renders correctly with options ["red", "yellow", "blue"]', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is big', async () => {
    wrapper.setProps({ big: true });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('when single selection, "yellow" is emitted', async () => {
    wrapper.setProps({ val: undefined });
    await wrapper.vm.$nextTick();
    wrapper
      .findAll('button')
      .at(1)
      .trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('input')).toBeTruthy();
    expect(wrapper.emitted('input')).toEqual([['yellow']]);
  });

  test('when multi selection, "[ yellow, blue ]" is emitted', async () => {
    wrapper
      .findAll('button')
      .at(1)
      .trigger('click');
    await wrapper.vm.$nextTick();
    wrapper
      .findAll('button')
      .at(2)
      .trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('input')).toBeTruthy();
    expect(wrapper.emitted('input')).toEqual([[['yellow']], [['blue']]]);
  });
});
