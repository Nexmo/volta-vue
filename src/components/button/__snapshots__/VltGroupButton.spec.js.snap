// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-group-button is big 1`] = `
<div
  class="Vlt-btn-group Vlt-btn-group--app Vlt-btn-group--big"
>
  <button
    class="Vlt-btn Vlt-btn--tertiary"
    value="red"
  >
    
    red
  
  </button>
  <button
    class="Vlt-btn Vlt-btn--tertiary"
    value="yellow"
  >
    
    yellow
  
  </button>
  <button
    class="Vlt-btn Vlt-btn--tertiary"
    value="blue"
  >
    
    blue
  
  </button>
</div>
`;

exports[`vlt-group-button renders correctly with options ["red", "yellow", "blue"] 1`] = `
<div
  class="Vlt-btn-group Vlt-btn-group--app"
>
  <button
    class="Vlt-btn Vlt-btn--tertiary"
    value="red"
  >
    
    red
  
  </button>
  <button
    class="Vlt-btn Vlt-btn--tertiary"
    value="yellow"
  >
    
    yellow
  
  </button>
  <button
    class="Vlt-btn Vlt-btn--tertiary"
    value="blue"
  >
    
    blue
  
  </button>
</div>
`;
